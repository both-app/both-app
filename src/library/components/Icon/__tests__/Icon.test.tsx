import React from 'react'
import { render } from 'react-native-testing-library'

import { Icon, IconName } from '../'

describe('Icon', () => {
  const ICONS = [
    ['arrow_right'],
    ['chevron_left'],
    ['check'],
    ['plus'],
    ['award'],
    ['heart'],
    ['list'],
    ['close'],
    ['share'],
    ['camera'],
    ['rotate'],
    ['loader'],
    ['linkedin'],
    ['users'],
  ] as Array<[IconName]>

  test.each(ICONS)('should display the icon', (iconName) => {
    const { toJSON } = render(<Icon iconName={iconName} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
