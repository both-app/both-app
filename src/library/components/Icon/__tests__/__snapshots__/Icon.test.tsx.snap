// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should display the icon 1`] = `<SvgMock />`;

exports[`Icon should display the icon 2`] = `<SvgMock />`;

exports[`Icon should display the icon 3`] = `<SvgMock />`;

exports[`Icon should display the icon 4`] = `<SvgMock />`;

exports[`Icon should display the icon 5`] = `<SvgMock />`;

exports[`Icon should display the icon 6`] = `<SvgMock />`;

exports[`Icon should display the icon 7`] = `<SvgMock />`;

exports[`Icon should display the icon 8`] = `<SvgMock />`;

exports[`Icon should display the icon 9`] = `<SvgMock />`;

exports[`Icon should display the icon 10`] = `<SvgMock />`;

exports[`Icon should display the icon 11`] = `<SvgMock />`;

exports[`Icon should display the icon 12`] = `<SvgMock />`;

exports[`Icon should display the icon 13`] = `<SvgMock />`;

exports[`Icon should display the icon 14`] = `<SvgMock />`;
